cpp/Platform.Equality/Platform.Equality.Any.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#ifndef PLATFORM_EQUALITY_ANY_H
+#define PLATFORM_EQUALITY_ANY_H
+
+#include <any>
+#include <functional>
+#include <typeindex>
+#include <concepts>
+#include <stdexcept>
+
+namespace Platform::Equality
+{
+    template<class T>
+    inline auto __to_any_equal_visitor(auto func)
+    {
+        return std::pair<std::type_index, std::function<bool(std::any, std::any)>>
+        {
+                std::type_index(typeid(T)),
+                [func](std::any a, std::any b) -> std::size_t
+                {
+                    if constexpr (std::is_void_v<T>)
+                        return func();
```

Comment:
Please as `{` and `}` brackets here (do no use single line if bodies). If later it will be required to extend such code or to add something for debug it will be much easier to maintain such code.